app/services/search/reading_list.rb

```diff
@@ -75,6 +86,8 @@ def self.find_articles(user:, term:, statuses:, tags:, page:, per_page:)
         "INNER JOIN (#{reaction_query_sql}) reactions ON reactions.reactable_id = articles.id",
       )
 
+      relation = relation.where(published: true) if only_published.present?
```

Comment:
Do we need `.present?` if this is not expected to treat empty arrays, strings, and hashes equivalently to `false` and `nil`?

Also, nitpick: [scope](https://github.com/forem/forem/blob/6e275a12fc5aa3db79887441cdbeee2901112032/app/models/article.rb#L180-L183). 😄 

```suggestion
      relation = relation.published if only_published
```